Shuttle-front/src/app/services/map/map-estimation.service.ts

```diff
@@ -0,0 +1,39 @@
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { ExpansionCase } from '@angular/compiler';
+import { Injectable } from '@angular/core';
+import { delay, Observable } from 'rxjs';
+import { environment } from 'src/environments/environment';
+
+export interface Location {
+  latitude: number,
+  longitude: number,
+  address: String,
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MapEstimationService {
+
+  constructor(private http: HttpClient) { 
+  }
+
+  search(place: String): Observable<any>{
+    return this.http.get(
+      'https://nominatim.openstreetmap.org/search?format=json&q=' + place
+    );
+  }
+
+  reverseSearch(lat: number, lon: number): Observable<any> {
+    return this.http.get(
+      `https://nominatim.openstreetmap.org/reverse?format=json&lat=${lat}&lon=${lon}&<params>`
+    );
+  }
+
+  getActiveDriversLocations(): Observable<Array<Location>> {
```

Comment:
This should **NOT** be in `MapEstimationService`, it's tied to `DriverService.java`, so it goes in `DriverService`!